service.py

```diff
@@ -128,13 +157,23 @@ def createListItemFromVideo(video):
     result = ydl.extract_info(url, download=False)
 
 if 'entries' in result:
-    # Playlist
+    # more than one video
     pl = xbmc.PlayList(1)
     pl.clear()
-    for video in result['entries']:
-        list_item = createListItemFromVideo(video);
-        xbmc.PlayList(1).add(list_item.getPath(), list_item)
-    xbmc.Player().play(pl)
+
+    # make sure first ListItem has a resolved url, to avoid recursion and Kodi crashes
+    firstVideoUrl = result['entries'][0]['url']
+    firstItem = createListItemFromVideo(ydl.extract_info(firstVideoUrl, download=False))
+    pl.add(firstItem.getPath(), firstItem)
+	
+    for video in result['entries'][1:]:
+        list_item = createListItemFromFlatPlaylistItem(video)
+        pl.add(list_item.getPath(), list_item)
+
+    #xbmc.Player().play(pl) # this probably works again
+    # ...but start playback the same way the Youtube plugin does it:
+    xbmc.executebuiltin('Playlist.PlayOffset(%s,%d)' % ('video', 0))
```

Comment:
xbmc.Player().play(pl) 
is from the official kodi python documentation
see the example here:
https://codedocs.xyz/AlwinEsch/kodi/group__python___player.html#ga1a8be149c2009e1da8577c649029ba30
i think this is the correct and official way to play a playlist.
while executebuiltin is like, run a script, or any internal command.

not the python expert, but:

how will this function even get the playlist and the list_item you created earlier?

Reply:
> how will this function even get the playlist and the list_item you created earlier?

First I thought you were asking how the plugin gets called back. If you're asking how Playlist.PlayOffset gets the playlist, then the answer is that "playlist" in Kodi lingo refers to the _play queue_, which is a global. No reference needed.